explainer.md

```diff
@@ -245,16 +436,35 @@ and Ubuntu don't support them at all).
+
+### Why can’t this be used in the background from the ServiceWorker? (see #28 and #5)
+
+Realistically, using the badge from the background is the most important use case, so ideally, badges could be updated in the background via the service worker.
+
+However there is a non-trivial problem: We need a way to inform the user that the app is running.
+
+With push notifications, browsers enforce that a notification be shown, or the browser will stop the app from processing additional notifications. Things are not as simple with badging, as a site could always set its badge to the same value, which would give the user would no cue that a site is running (or has run) in the background (and could mine all the cryptocurrencies!).
 
-Ideally, it would be, and we're considering this for a future version of this API. 
-However, the API will require some more thought, as there can be multiple apps installed
-for a single service worker, so we would need some way of specifying which one should
-be badged.
+To solve this, we are considering a separate notification channel especially for badges. The browser would know how to turn the notification into a badge without running any JavaScript.
+
+> Note: I might be pretty far off the mark here, I’m not that familiar in things service worker, background, or notifications.
+
+### Is this API useful for mobile OS’s?
+iOS has support for badging APIs (see [iOS](#ios). However, PWAs on iOS will not be able to use the badging API until Safari (the only browser engine allowed on iOS) adds support for it.
+
+On Android, badging is blocked at an OS level, as there is [no API for setting a badge without also displaying a notification](#android). However, a badge will already be displayed if a PWA has pending notifications (it just doesn’t allow the fine grained control proposed by this API).
+
+To summarize: This API cannot be used in PWAs on either mobile operating system. Support on iOS is blocked until Safari implements the API and Android does not have an API for controlling badges. Should either situation change, the badging API will become trivially available.
+
+### Why is this API attached to `window` instead of `navigator` or `notifications`?
+
+There was a [poll](https://github.com/WICG/badging/issues/14#issuecomment-445548190), and the current style seemed most popular. There is more detail and discussion in [Issue 14](https://github.com/WICG/badging/issues/14).
 
 ### Is there an upper limit on the size of the integer? And if so, what's the behavior if that limit is reached?
 
-There is no upper limit (besides 2<sup>31</sup>). However, each user agent is
+There is no upper limit (besides 2<sup>64</sup>). However, each user agent is
```

Comment:
Why 2^64? I believe 2^32 is the largest value that can be stored in a WebIDL long which is the type we're using here (any value larger will saturate or something).

Since JS numbers are stored in a double, the limit wouldn't be 2^64, it would be 2^53 (precisely) or 2^2047 (imprecisely), but that's irrelevant since it gets truncated to a long.

Reply:
I think the [WebIDL](https://cs.chromium.org/chromium/src/third_party/blink/renderer/modules/badging/experimental_badge.idl?type=cs&q=f:experimental_badge.idl&sq=package:chromium&g=0&l=15) at the moment is taking an `unsigned long long`, which is a `uint64_t` in C++. Not sure about how the translation works?

Reply:
Oh, now I remember. This was requested on #21 (which I didn't really want to do, but apparently it's convention).

[Per WebIDL](https://heycam.github.io/webidl/#abstract-opdef-converttoint), a JS conversion to long long with `[EnforceRange]` is clamped at 2^53, so we don't end up rounding integers.

So change this to 2^53.

(Aside, I'm curious why it's 2^53 - 1 and not just 2^53. 2^53 is the largest integer that can be represented in a double along with all smaller integers, so it seems like just a mistake to not allow it.)

Reply:
I've updated this to be `Number.MAX_SAFE_INTEGER`.

Reply:
Oh, nice find. I didn't know about that.